sdcm/nemesis.py

```diff
@@ -3820,6 +3820,108 @@ def _shrink_cluster(self, rack=None):
+        # TODO: update the 'kms_host' parameter when  dynamic key creation gets implemented.
+        self._enable_disable_table_encryption(additional_scylla_encryption_options={
+            'key_provider': 'KmsKeyProviderFactory', 'kms_host': 'kms_test'})
+
+    def _enable_disable_table_encryption(self, additional_scylla_encryption_options=None):
+        if additional_scylla_encryption_options and additional_scylla_encryption_options.get(
+                'key_provider', 'N/A') == 'KmsKeyProviderFactory':
+            for node in self.cluster.nodes:
+                is_restart_needed = False
+                with node.remote_scylla_yaml() as scylla_yml:
+                    if not scylla_yml.kms_hosts:
+                        scylla_yml.kms_hosts = {}
+                    # TODO: update the 'kms_hosts' field when the dynamic key creation gets implemented.
+                    if 'kms_test' not in scylla_yml.kms_hosts:
+                        scylla_yml.kms_hosts['kms_test'] = {
+                            'master_key': 'alias/kms_encryption_test',
+                            'aws_region': 'us-east-1',
+                            'aws_use_ec2_credentials': True,
+                        }
+                        is_restart_needed = True
+                if is_restart_needed:
+                    node.configure_kms()
+                    node.process_scylla_args('--logger-log-level kms=trace')
+                    node.restart_scylla()
+
+        # Create table with encryption
+        keyspace_name, table_name = self.cluster.get_test_keyspaces()[0], 'tmp_encrypted_table'
+        scylla_encryption_options = {'cipher_algorithm': 'AES/ECB/PKCS5Padding', 'secret_key_strength': 128}
+        scylla_encryption_options |= additional_scylla_encryption_options or {}
+        self.log.info("Create '%s.%s' table with server encryption", keyspace_name, table_name)
+        with self.cluster.cql_connection_patient(self.target_node, keyspace=keyspace_name) as session:
+            # NOTE: scylla-bench expects following table structure:
+            #       (pk bigint, ck bigint, v blob, PRIMARY KEY(pk, ck)) WITH compression = { }
+            create_table_query_cmd = (
+                f"CREATE TABLE IF NOT EXISTS {table_name}"
+                " (pk bigint, ck bigint, v blob, PRIMARY KEY (pk, ck))"
+                " WITH compression = { } AND read_repair_chance=0.0"
+                f" AND compaction = {{ 'class' : '{self.cluster.params.get('compaction_strategy')}' }}"
+                f" AND scylla_encryption_options = {scylla_encryption_options};")
+            session.execute(create_table_query_cmd)
+
+        def upgrade_sstables(nodes):
+            for node in nodes:
+                self.log.info("Upgradesstables on the '%s' node for the new encrypted table", node.name)
+                node.remoter.run(f'nodetool upgradesstables -- {keyspace_name} {table_name}', verbose=True)
```

Comment:
@elcallio @eliransin 

isn't upgradesstable needs to be called with `--include-all-sstables` ? to force a rewrite on all sstable ?
it's weird the docs doesn't mention it in: https://opensource.docs.scylladb.com/stable/operating-scylla/security/encryption-at-rest.html

Reply:
If I understand the docs [1] correctly then in our case it will not make any difference because our target to get the desired state. And this `desired state` will be achieved in both cases.

[1] https://opensource.docs.scylladb.com/stable/operating-scylla/nodetool-commands/upgradesstables.html

Reply:
The docs talks mainly about sstable format version (I have tested it when it was introduced, no other state was defined back then)

We'll see, also the relevent code in SCT currently uses this flag.

Reply:
Yes, should include a `-a`. Docs should reflect this. I am pretty sure there are docs that does, even though the linked one above does not.

Reply:
Done.